doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
 
 See the `Developer Workflow`_ instructions below for setting up a github
 account, signing the contributor agreement, and submitting pull requests.
+This document contains a mixture of tips for git beginners and specific
+Chapel workflow recommendations.
 
 Developer Workflow
 ------------------
 
 Overview:
 
-#. `Fork repo on github`_
-#. `Create new branch`_
-#. `Submit pull request`_
+#. `Discuss design`_ changes or big development efforts
+#. `Set up a branch for development`_
+
+   #. `Set up repository`_
+   #. `Create new branch`_
+
+#. `Develop and test contributions locally`_
+
+   #. `Test your feature`_
+
+#. `Request feedback on your changes`_
+
+   #. `Push your work`_
+   #. `Ask for feedback early`_ (optional)
+   #. `Submit pull request`_
+   #. `Find a reviewer`_
+   #. `Work with your reviewers`_
+
 #. `Test changes`_ (can be done by team at Cray)
```

Comment:
I think we should remove "can be done by team at Cray". Also, the original outline (which is in your PR message) doesn't have this as a top-level item, and instead lists it under a checklist of things to do before merging. I think I'd still be in favor of moving it under Merge changes to master or renaming the section "Getting Ready to Merge" and include it as part of a checklist. The full checklist is in your PR message. I think I'd prefer the checklist exist in outline form at the start of "Getting Ready to Merge". Most of it is covered currently in the long-sentences form, but a checklist can also be a quick guide.